pkg/apis/v1/nodepool_conversion.go

```diff
@@ -0,0 +1,167 @@
+	"knative.dev/pkg/apis"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+)
+
+// Convert v1 NodePool to v1beta1 NodePool
+func (in *NodePool) ConvertTo(ctx context.Context, to apis.Convertible) error {
+	v1beta1NP := to.(*v1beta1.NodePool)
+	v1beta1NP.Name = in.Name
+	v1beta1NP.UID = in.UID
+	v1beta1NP.Labels = in.Labels
+	v1beta1NP.Annotations = in.Annotations
+	in.Spec.convertTo(ctx, &v1beta1NP.Spec)
+
+	// Convert v1 status
+	v1beta1NP.Status.Resources = in.Status.Resources
+
+	v1beta1NP.Annotations = lo.Assign(v1beta1NP.Annotations, map[string]string{
+		NodePoolHashVersion:              v1beta1NP.Hash(),
+		NodePoolHashVersionAnnotationKey: v1beta1.NodePoolHashVersion,
+	})
+
+	return nil
+}
+
+func (in *NodePoolSpec) convertTo(ctx context.Context, v1beta1np *v1beta1.NodePoolSpec) {
+	v1beta1np.Weight = in.Weight
+	v1beta1np.Limits = v1beta1.Limits(in.Limits)
+	in.Disruption.convertTo(&v1beta1np.Disruption)
+	in.Template.convertTo(ctx, &v1beta1np.Template)
+}
+
+func (in *Disruption) convertTo(v1beta1np *v1beta1.Disruption) {
+	v1beta1np.ConsolidateAfter = (*v1beta1.NillableDuration)(in.ConsolidateAfter)
+	v1beta1np.ConsolidationPolicy = v1beta1.ConsolidationPolicy(in.ConsolidationPolicy)
+	v1beta1np.ExpireAfter = v1beta1.NillableDuration(in.ExpireAfter)
+	v1beta1np.Budgets = lo.Map(in.Budgets, func(v1Budget Budget, _ int) v1beta1.Budget {
+		return v1beta1.Budget{
+			Nodes:    v1Budget.Nodes,
+			Schedule: v1Budget.Schedule,
+			Duration: v1Budget.Duration,
+		}
+	})
+}
+
+func (in *NodeClaimTemplate) convertTo(ctx context.Context, v1beta1np *v1beta1.NodeClaimTemplate) {
+	v1beta1np.Annotations = in.Annotations
+	v1beta1np.Labels = in.Labels
+	v1beta1np.Spec.Taints = in.Spec.Taints
+	v1beta1np.Spec.StartupTaints = in.Spec.StartupTaints
+	v1beta1np.Spec.Requirements = lo.Map(in.Spec.Requirements, func(v1Requirements NodeSelectorRequirementWithMinValues, _ int) v1beta1.NodeSelectorRequirementWithMinValues {
+		return v1beta1.NodeSelectorRequirementWithMinValues{
+			NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				Key:      v1Requirements.Key,
+				Values:   v1Requirements.Values,
+				Operator: v1Requirements.Operator,
+			},
+			MinValues: v1Requirements.MinValues,
+		}
+	})
+
+	if in.Spec.NodeClassRef != nil {
+		nodeclass, found := lo.Find(injection.GetNodeClasses(ctx), func(nc schema.GroupVersionKind) bool {
+			return nc.Kind == in.Spec.NodeClassRef.Kind && nc.Group == in.Spec.NodeClassRef.Group
+		})
+		v1beta1np.Spec.NodeClassRef = &v1beta1.NodeClassReference{
+			Kind:       in.Spec.NodeClassRef.Kind,
+			Name:       in.Spec.NodeClassRef.Name,
+			APIVersion: lo.Ternary(found, nodeclass.GroupVersion().String(), ""),
+		}
+	}
+
+	// Need to implement Kubelet Conversion
+}
+
+// Convert v1beta1 NodePool to V1 NodePool
+func (in *NodePool) ConvertFrom(ctx context.Context, v1beta1np apis.Convertible) error {
+	v1beta1NP := v1beta1np.(*v1beta1.NodePool)
+	in.Name = v1beta1NP.Name
```

Comment:
Same comment here. How do we make sure that we don't lose our ObejctMeta that isn't included here?

Reply:
Is it fine to just set in.Meta to v1beta1NP.Meta?

Reply:
Yes, it is. I was trying to be careful with how we handled drift, so I has just moving one component at a time. With our new approach, I will be just copying over the fields